File: DristorApp/Data/Models/OrderItem.cs
using System;
namespace DristorApp.Data.Models
{
    public class OrderItem
    {
        public int Id { set; get; }
        public int Quantity { set; get; }

        public int? CouponId { set; get; }
        public virtual Coupon Coupon { set; get; }
        public virtual ProductVariant ProductVariant { set; get; }
        public virtual Order Order { set; get; }
    }
}

File: frontend/src/app/models/order-item.ts
import {IProductVariant} from "./product-variant";
import {ICoupon} from "./coupon";
import {IOrder} from "./order";

export interface IOrderItem {
  id: number;
  quantity: number;

  coupon?: ICoupon;
  productVariant?: IProductVariant;
  order?: IOrder;
}

export interface IOrderItemCreate {
  quantity: number;

  productVariant: number;
  order: number;
}
